src/queries/login.ts

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react';
+
+export type LoginInfo =
+  | { type: 'LOADING' }
+  | { type: 'NOT_LOGGED_IN' }
+  | { type: 'LOGGED_IN'; name: string };
+
+export function useLogin(): LoginInfo {
+  const [loginInfo, setLoginInfo] = useState<LoginInfo>({ type: 'LOADING' });
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    fetch("EPH-60のエンドポイントURL", { signal })
+      .then(response => response.json())
+      .then(data => {
+        if (data.isLoggedIn) {
+          setLoginInfo({ type: 'LOGGED_IN', name: data.name });
+        } else {
+          setLoginInfo({ type: 'NOT_LOGGED_IN' });
+        }
+      })
```

Comment:
```suggestion
    (async () => {
      try {
        const response = await fetch("EPH-60のエンドポイントURL", { signal });
        if (!response.ok) {
          setLoginInfo({ type: 'NOT_LOGGED_IN' });
          return;
        }
        const { name } = await response.json();      
        setLoginInfo({ type: 'LOGGED_IN', name });
      } catch {
        setLoginInfo({ type: 'NOT_LOGGED_IN' });
      }
    })();
```
ちょっと処理フローが `then` の連鎖で書くにはつらいので, `async` 関数を作って即実行するような具合でどうでしょう.
認証していない場合はレスポンスが 2xx 以外 (`response.ok` が `false`) になって JSON が返ってこないので, `response.json()` より前に分岐することになります.
なお, `setLoginInfo({ type: 'NOT_LOGGED_IN' });` を 2 つ別々に書いていますが, 最初の方は認証失敗によるものですが, `catch` 内のほうはネットワークエラー (通信切断など) によるものなので少し状況が異なります. 要望によっては今後リトライ処理を入れるかもしれないのでこうしておきたいです.